### PR TITLE
Update after recent OpenTracing-java changes

### PR DESCRIPTION
**Pull request by bensigelman**
Make adjustments per https://github.com/opentracing/opentracing-java/pull/33

cc: @bcronin @kcamenzind

### CODE REVIEW

File: common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java [hunk 1]
```diff
@@ -293,29 +296,29 @@ public void disable() {
...
-  public void inject(io.opentracing.SpanContext spanContext, Object carrier) {
+  public <C> void inject(io.opentracing.SpanContext spanContext, Format<C> format, C carrier) {
     SpanContext lightstepSpanContext = (SpanContext)spanContext;
-    if (Propagator.TEXT_MAP.matchesInjectCarrier(carrier)) {
-      Propagator.TEXT_MAP.inject(lightstepSpanContext, carrier);
-    } else if (Propagator.HTTP_HEADER.matchesInjectCarrier(carrier)) {
-      Propagator.HTTP_HEADER.inject(lightstepSpanContext, carrier);
-    } else if (Propagator.BINARY.matchesInjectCarrier(carrier)) {
+    if (format == Format.Builtin.TEXT_MAP) {
+      Propagator.TEXT_MAP.inject(lightstepSpanContext, (TextMap)carrier);
```
Comment (bensigelman) [inline common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java]: these casts are a cheesy way of doing this, but building a full-on registry for two types seems ~ridiculous...

File: common/src/main/java/com/lightstep/tracer/shared/Propagator.java [hunk 2]
```diff
@@ -86,46 +76,24 @@ public SpanContext extract(Object carrier) {
...
-    }
-    public boolean matchesExtractCarrier(Object carrier) {
-      return carrier instanceof HttpHeaderReader;
+  static final Propagator<TextMap> HTTP_HEADERS = new Propagator<TextMap>() {
+    public void inject(SpanContext spanContext, TextMap carrier) {
+      TEXT_MAP.inject(spanContext, carrier);
     }
-    public void inject(SpanContext spanContext, Object carrier) {
-      final HttpHeaderWriter headerWriter = (HttpHeaderWriter)carrier;
-      TEXT_MAP.inject(spanContext, new TextMapWriter() {
```
Comment (bensigelman) [inline common/src/main/java/com/lightstep/tracer/shared/Propagator.java]: boilerplate reduction!

File: common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java [hunk 3]
```diff
@@ -293,29 +296,29 @@ public void disable() {
     return this.new SpanBuilder(operationName);
   }
 
-  public void inject(io.opentracing.SpanContext spanContext, Object carrier) {
+  public <C> void inject(io.opentracing.SpanContext spanContext, Format<C> format, C carrier) {
```
Comment (bensigelman) [inline common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java]: the interface change... Format is now an explicit type and param.

**Comment by bcronin** [comment]
The `jre-simple` example calls inject and needs to be updated to reflect the changes as well.  (Run `make test` or, more narrowly, `make -C examples/jre-simple run`)

**Comment by bensigelman** [comment]
@bcronin thanks -- I ran the tests in the jre and android dirs but forgot about the examples :-/

**Pull request merged**